### PR TITLE
EOS-26270 : [F-53B]Support for core dump include/exclude support-bundle filter

### PR DESCRIPTION
**Pull request by papan-singh**
…e filter.

Parsed MOTR_M0D_DATA_DIR from /etc/sysconfig/motr and added to CORE_DIRS directory to collect core.

Signed-off-by: Papan Kumar Singh <papan.kumarsingh@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by papan-singh** [comment]
Patch is tested on k8s env

1 . coredumps false
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 /]# /opt/seagate/cortx/motr/libexec/m0reportbug-bundler SBbw5f7n9i /var/log/support_bundle --duration "2020-09-06T05:30:00P5DT3H3S" --size_limit 100MB --services All --binlogs false --coredumps false --stacktrace false
probes... OK
misc_files... OK
backtraces_live... OK
coredump_live... WARNING ## live core are not collected on kvm ##
OK
backtraces_cores... OK
m0traces... OK
addb... OK
addb-files... OK
systemd_journal... No journal files were found.
OK
sysrq_info... WARNING /proc/sysrq-trigger is read-only. Won't be able to collect system information.
OK
WARNING /var/crash/: no such directory
WARNING /var/log/crash/: no such directory
WARNING /var/log/seagate/motr/tmpspace/: no such directory
m0traces /opt/seagate/cortx/motr/libexec... OK
m0traces /etc/cortx/log/motr/17b20f538919452b9b2fc431b078b6be/trace... OK (1m23s)
addb-files /etc/cortx/log/motr/17b20f538919452b9b2fc431b078b6be/addb... OK (1m21s)
Please attach these files to the bug report:
m0reportbug-data.tar.gz
m0reportbug-cores.tar.gz
m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 /]# cd /var/log/support_bundle/motr/
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# ls
motr_SBbw5f7n9i.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# tar -xvf motr_SBbw5f7n9i.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1830/
cortx-data-headless-svc-ssc-vm-rhev4-1830/_m0reportbug-bundler.stderr
cortx-data-headless-svc-ssc-vm-rhev4-1830/m0reportbug-data.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1830/m0reportbug-cores.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1830/m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# ls
cortx-data-headless-svc-ssc-vm-rhev4-1830 motr_SBbw5f7n9i.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# cd cortx-data-headless-svc-ssc-vm-rhev4-1830/
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 cortx-data-headless-svc-ssc-vm-rhev4-1830]# ls
_m0reportbug-bundler.stderr m0reportbug-cores.tar.gz m0reportbug-data.tar.gz m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 cortx-data-headless-svc-ssc-vm-rhev4-1830]# tar -xvf m0reportbug-cores.tar.gz
m0reportbug-cores/
m0reportbug-cores/var/
m0reportbug-cores/var/motr/
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 cortx-data-headless-svc-ssc-vm-rhev4-1830]#

2. coredumps true
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 ~]# /opt/seagate/cortx/motr/libexec/m0reportbug-bundler SBbw5f7n9i /var/log/support_bundle --duration "2020-09-06T05:30:00P5DT3H3S" --size_limit 100MB --services All --binlogs false --coredumps true --stacktrace false
probes... OK
misc_files... OK
backtraces_live... OK
coredump_live... WARNING ## live core are not collected on kvm ##
OK
backtraces_cores... OK
m0traces... OK
addb... OK
addb-files... OK
systemd_journal... No journal files were found.
OK
sysrq_info... WARNING /proc/sysrq-trigger is read-only. Won't be able to collect system information.
OK
WARNING /var/crash/: no such directory
WARNING /var/log/crash/: no such directory
backtraces_core /etc/cortx/motr... OK (1m4s)
WARNING /var/log/seagate/motr/tmpspace/: no such directory
m0traces /opt/seagate/cortx/motr/libexec... OK
m0traces /etc/cortx/log/motr/17b20f538919452b9b2fc431b078b6be/trace... OK (1m28s)
addb-files /etc/cortx/log/motr/17b20f538919452b9b2fc431b078b6be/addb... OK (1m22s)
Please attach these files to the bug report:
m0reportbug-data.tar.gz
m0reportbug-cores.tar.gz
m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 ~]# cd /var/log/support_bundle/motr/
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# ls
motr_SBbw5f7n9i.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# tar -xvf motr_SBbw5f7n9i.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1830/
cortx-data-headless-svc-ssc-vm-rhev4-1830/_m0reportbug-bundler.stderr
cortx-data-headless-svc-ssc-vm-rhev4-1830/m0reportbug-data.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1830/m0reportbug-cores.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1830/m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# ls
cortx-data-headless-svc-ssc-vm-rhev4-1830 motr_SBbw5f7n9i.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# cd cortx-data-headless-svc-ssc-vm-rhev4-1830/
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 cortx-data-headless-svc-ssc-vm-rhev4-1830]# ls
_m0reportbug-bundler.stderr m0reportbug-cores.tar.gz m0reportbug-data.tar.gz m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 cortx-data-headless-svc-ssc-vm-rhev4-1830]# tar -xvf m0reportbug-cores.tar.gz
m0reportbug-cores/
m0reportbug-cores/var/
m0reportbug-cores/var/motr/
m0reportbug-cores/etc/
m0reportbug-cores/etc/cortx/
m0reportbug-cores/etc/cortx/motr/
m0reportbug-cores/etc/cortx/motr/m0d-0x7200000000000001:0xa/
m0reportbug-cores/etc/cortx/motr/m0d-0x7200000000000001:0xa/core.94.xz

File: utils/m0reportbug [hunk 1]
```diff
@@ -886,6 +894,20 @@ optparse() {
             --addb) COLLECT_ADDB_P=1;;
             --skip-addb-files) SKIP_ADDB_FILE_P=1;;
             -b|--binaries) COLLECT_BINARIES_P=1;;
+            --coredumps) 
+                if [ -n "$2" ];
```
Comment (nkommuri) [inline utils/m0reportbug:898]: As discussed, get rid off this condition and validate --coredump option to be either "true" or "false".

Comment (papan-singh) [reply utils/m0reportbug]: incorporated

File: utils/m0reportbug [hunk 2]
```diff
@@ -886,6 +895,20 @@ optparse() {
...
+                else
+                    echo "$2 reqire argument" && usage
+                fi
+                if [ $coredump == "true" ];
+                then
+                    COLLECT_COREDUMPS_P=1
+                else
+                    COLLECT_COREDUMPS_P=0
+                fi
+                shift;;
```
Comment (yatin-mahajan) [inline utils/m0reportbug:911]: Use "--coredumps" option  in same fashion  as "--binaries, --addb, --skip-addb-files" are used in m0reportbug. 

Simply set "OLLECT_COREDUMPS_P=1" if "--coredumps" is specified on command line, do not check "true || false"

**Comment by papan-singh** [comment]
tested latest patch in k8s env
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 /]# /opt/seagate/cortx/motr/libexec/m0reportbug-bundler SBbw5f7n9i /var/log/support_bundle --duration "2020-09-06T05:30:00P5DT3H3S" --size_limit 100MB --services All --binlogs false --coredumps true --stacktrace false
probes... OK
misc_files... OK
backtraces_live... OK
coredump_live... *WARNING* ## live core are not collected on kvm ##
OK
backtraces_cores... OK
m0traces... OK
addb... OK
addb-files... OK
systemd_journal... No journal files were found.
OK
sysrq_info... *WARNING* /proc/sysrq-trigger is read-only. Won't be able to collect system information.
OK
*WARNING* /var/crash/: no such directory
*WARNING* /var/log/crash/: no such directory
backtraces_core /etc/cortx/motr... OK (0m35s)
*WARNING* /var/log/seagate/motr/tmpspace/: no such directory
m0traces /opt/seagate/cortx/motr/libexec... OK
m0traces /etc/cortx/log/motr/17b20f538919452b9b2fc431b078b6be/trace... OK (1m0s)
addb-files /etc/cortx/log/motr/17b20f538919452b9b2fc431b078b6be/addb... OK (2m28s)
Please attach these files to the bug report:
  m0reportbug-data.tar.gz
  m0reportbug-cores.tar.gz
  m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 /]# cd /var/log/support_bundle/motr/
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# ls
motr_SBbw5f7n9i.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# tar -xvf motr_SBbw5f7n9i.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1830/
cortx-data-headless-svc-ssc-vm-rhev4-1830/_m0reportbug-bundler.stderr
cortx-data-headless-svc-ssc-vm-rhev4-1830/m0reportbug-data.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1830/m0reportbug-cores.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1830/m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# ls
cortx-data-headless-svc-ssc-vm-rhev4-1830  motr_SBbw5f7n9i.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# cd cortx-data-headless-svc-ssc-vm-rhev4-1830/
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 cortx-data-headless-svc-ssc-vm-rhev4-1830]# ls
_m0reportbug-bundler.stderr  m0reportbug-cores.tar.gz  m0reportbug-data.tar.gz  m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 cortx-data-headless-svc-ssc-vm-rhev4-1830]# tar -xvf m0reportbug-cores.tar.gz
m0reportbug-cores/
m0reportbug-cores/var/
m0reportbug-cores/var/motr/
m0reportbug-cores/etc/
m0reportbug-cores/etc/cortx/
m0reportbug-cores/etc/cortx/motr/
m0reportbug-cores/etc/cortx/motr/m0d-0x7200000000000001:0xa/
m0reportbug-cores/etc/cortx/motr/m0d-0x7200000000000001:0xa/core.94.xz


[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 ~]# /opt/seagate/cortx/motr/libexec/m0reportbug-bundler SBbw5f7n9i /var/log/support_bundle --duration "2020-09-06T05:30:00P5DT3H3S" --size_limit 100MB --services All --binlogs false --coredumps false --stacktrace false
probes... OK
misc_files... OK
backtraces_live... OK
coredump_live... *WARNING* ## live core are not collected on kvm ##
OK
backtraces_cores... OK
m0traces... OK
addb... OK
addb-files... OK
systemd_journal... No journal files were found.
OK
sysrq_info... *WARNING* /proc/sysrq-trigger is read-only. Won't be able to collect system information.
OK
*WARNING* /var/crash/: no such directory
*WARNING* /var/log/crash/: no such directory
*WARNING* /var/log/seagate/motr/tmpspace/: no such directory
m0traces /opt/seagate/cortx/motr/libexec... OK
m0traces /etc/cortx/log/motr/17b20f538919452b9b2fc431b078b6be/trace... OK (0m59s)
addb-files /etc/cortx/log/motr/17b20f538919452b9b2fc431b078b6be/addb... OK (2m20s)
Please attach these files to the bug report:
  m0reportbug-data.tar.gz
  m0reportbug-cores.tar.gz
  m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 ~]# cd /var/log/support_bundle/motr/
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# ls
motr_SBbw5f7n9i.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# tar -xvf motr_SBbw5f7n9i.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1830/
cortx-data-headless-svc-ssc-vm-rhev4-1830/_m0reportbug-bundler.stderr
cortx-data-headless-svc-ssc-vm-rhev4-1830/m0reportbug-data.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1830/m0reportbug-cores.tar.gz
cortx-data-headless-svc-ssc-vm-rhev4-1830/m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# ls
cortx-data-headless-svc-ssc-vm-rhev4-1830  motr_SBbw5f7n9i.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 motr]# cd cortx-data-headless-svc-ssc-vm-rhev4-1830/
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 cortx-data-headless-svc-ssc-vm-rhev4-1830]# ls
_m0reportbug-bundler.stderr  m0reportbug-cores.tar.gz  m0reportbug-data.tar.gz  m0reportbug-traces.tar.gz
[root@cortx-data-headless-svc-ssc-vm-rhev4-1830 cortx-data-headless-svc-ssc-vm-rhev4-1830]# tar -xvf m0reportbug-cores.tar.gz
m0reportbug-cores/
m0reportbug-cores/var/
m0reportbug-cores/var/motr/

**Review by VinothVijayakumar-dev — APPROVED** [review]

**Review by yatin-mahajan — APPROVED** [review]
Looks good to me.

**Pull request merged**